### ISSUE TITLE
TS 26.532: Mismatch between normative specification of DataReportingSession maps and OpenAPI YAML

### ISSUE THREAD

**Issue by davidjwbbc** [5G-MAG/Standards#116]
# Context
TS 26.532 V17.2.0 and V18.1.0:
- Table 7.3.2.1-1 describes the properties _samplingRules_, _reportingConditions_ and _reportingRules_ of _DataReportingSession_ as maps from a _DataDomain_ to an array of objects relating to the property.
- Table 7.3.3.1-1 describes DataDomain as a simple enumeration.
- Clause B.4 presents the OpenAPI YAML representing _DataReportingSession_ and _DataDomain_.
  - _DataDomain_ contains the enumeration, and as is usual with 3GPP specifications there is a future expansion fallback to any string.
  - _DataReportingSession_ contains arrays of objects for _samplingRules_, _reportingConditions_ and _reportingRules_ instead of maps. Each object contains a pair of properties, the _DataDomain_ and a property containing the relevant object type.

# Problem
The YAML schema for _DataReportingSession_ does not match map type as given in table 7.3.2.1-1. Instead, the table uses a map and the YAML uses an array of 2-tuple like objects.

The arrays defined in the YAML cannot restrict each entry having a unique _DataDomain_ key as is implied by the map type from the table.

# Suggested Solution
Since _DataDomain_ is also defined as any string (future-proofing fallback with specific string values defined in TS 26.532 as the enumeration), each of the map properties in the table would therefore be mapping any string to an array of objects.

The mapping of a string to a value type perfectly describes an associative array mapping, which is better represented in OpenAPI by using an object (i.e. `type: object`) with `additionalProperties` for the value type of the map. This would also enforce the uniqueness of the _DataDomain_ key to the map and provide a truer reflection of the definition in the table.

Therefore the OpenAPI YAML for _DataReportingSession_ would become:
```yaml
components:
  schemas:
    DataReportingSession:
      description: "A representation of a Data Reporting Session."
      type: object
      properties:
        sessionId:
          $ref: 'TS26512_CommonData.yaml#/components/schemas/ResourceId'
        validUntil:
          $ref: 'TS29571_CommonData.yaml#/components/schemas/DateTime'
        externalApplicationId:
          $ref: 'TS29571_CommonData.yaml#/components/schemas/ApplicationId'
        supportedDomains:
          type: array
          items:
            $ref: '#/components/schemas/DataDomain'
          minItems: 0
        samplingRules:
          type: object
          additionalProperties:
              type: array
              items:
                $ref: 'TS26532_CommonData.yaml#/components/schemas/DataSamplingRule'
              minItems: 0
        reportingConditions:
          type: object
          additionalProperties:
              type: array
              items:
                $ref: '#/components/schemas/ReportingCondition'
              minItems: 0
        reportingRules:
          type: object
          additionalProperties:
              type: array
              items:
                $ref: 'TS26532_CommonData.yaml#/components/schemas/DataReportingRule'
              minItems: 0
      required:
        - externalApplicationId
        - supportedDomains
        - reportingConditions
```

**Comment by rjb1000** [comment]
Given that this issue affects the reporting of UE data to the standalone Data Collection AF (i.e. for exposure of the basic SA2 events to NWDAF), I think changes to both Rel-17 and Rel-18 are justified, even though 5G-MAG Reference Tools is only implementing Rel-18.

**Comment by rjb1000** [comment]
Change Requests agreed without revision at SA4#127-bis-e:
* TS 26.532
  * Rel-17 **CR0007**r2 "[EVEX] Essential maintenance" in [S4-240553](https://portal.3gpp.org/DesktopModules/CRs/CrDetails.aspx?CrId=524739).
  * Rel-18 **CR0009** "[EVEX, TEI18] Essential maintenance" in [S4-240554](https://portal.3gpp.org/DesktopModules/CRs/CrDetails.aspx?CrId=524740).

These will go to SA#104 for approval in June.

**Comment by rjb1000** [comment]
New specification versions published following approval of CRs at SA#104:
- TS 26.532
  - Rel-17: [V17.3.0](https://www.3gpp.org/ftp/Specs/archive/26_series/26.532/26532-h30.zip).
  - Rel-18: [V18.2.1](https://www.3gpp.org/ftp/Specs/archive/26_series/26.532/26532-i21.zip).
